test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -152,6 +157,142 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
+			gomega.Expect(k8sClient.Create(ctx, prodWl2)).Should(gomega.Succeed())
+
+			ginkgo.By("create the 'prod3' workload after a second")
+			fakeClock.Sleep(time.Second)
+			prodWl3 := testing.MakeWorkload("prod3", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl3)).Should(gomega.Succeed())
+
+			ginkgo.By("checking the 'prod1' workload is admitted and the 'prod2' workload is waiting")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl1)
+			util.ExpectWorkloadsToBeWaiting(ctx, k8sClient, prodWl2)
+
+			ginkgo.By("exceed the timeout for the 'prod1' workload")
+			fakeClock.Sleep(3 * time.Minute)
+
+			ginkgo.By("wait for the 'prod1' workload to be unadmitted")
+			gomega.Eventually(func() *kueue.Admission {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodWl1), prodWl1)).Should(gomega.Succeed())
+				return prodWl1.Spec.Admission
+			}, util.Timeout, util.Interval).Should(gomega.BeNil())
+
+			ginkgo.By("verify the 'prod2' workload gets admitted and the 'prod1' is waiting")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl2)
+			util.ExpectWorkloadsToBeWaiting(ctx, k8sClient, prodWl1)
+		})
+
+		ginkgo.It("Should unblock admission of new workloads once the admitted workload exceeds timeout", func() {
+			ginkgo.By("create the 'prod' workload")
+			prodWl := testing.MakeWorkload("prod", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl)).Should(gomega.Succeed())
+
+			ginkgo.By("create the 'dev' workload after a second")
+			fakeClock.Sleep(time.Second)
+			devWl := testing.MakeWorkload("dev", ns.Name).Queue(devQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, devWl)).Should(gomega.Succeed())
+
+			ginkgo.By("wait for the 'prod' workload to be admitted")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl)
+
+			ginkgo.By("verify the 'prod' queue resources are used")
+			gomega.Eventually(func() kueue.ClusterQueueStatus {
+				var updatedCQ kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodClusterQ), &updatedCQ)).To(gomega.Succeed())
+				return updatedCQ.Status
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(kueue.ClusterQueueStatus{
+				PendingWorkloads:  0,
+				AdmittedWorkloads: 1,
+				UsedResources: kueue.UsedResources{
+					corev1.ResourceCPU: {
+						"default": {
+							Total: pointer.Quantity(resource.MustParse("2")),
+						},
+					},
+				},
+				Conditions: []metav1.Condition{
+					{
+						Type:    kueue.ClusterQueueActive,
+						Status:  metav1.ConditionTrue,
+						Reason:  "Ready",
+						Message: "Can admit new workloads",
+					},
+				},
+			}, ignoreCQConditionTimestamps))
+
+			ginkgo.By("wait for the timeout to be exceeded")
```

Comment:
wouldn't the workload `dev` be admitted immediately?

Reply:
No, becuase 'dev' is waiting on the `PodsReady=True` in 'prod', which is never set in the test. Expanded the step description and added an explicit assert.

Reply:
In that case, I don't think this test case is different from the previous one. Why not add the check about usage to the previous test?

Reply:
In the previous test the prod cluster queue is used by both workloads, so the quota as released for a very short time as it gets immediately taken by the other workload.